projects/hslayers/src/hslayers.component.ts

```diff
@@ -1,117 +1,181 @@
 
-  /**
-   * Check if panel is configured to be visible in hsConfig.panelsEnabled
-   * or hsLayoutService.panelsEnabledDefaults and create one if so.
-   * @param name - Name of panel used in panelsEnabled config
-   * @param panelComponent - Class defining panel
-   * @param data - Extra misc data object to be stored in panel
-   */
-  createPanel(name: string, panelComponent: Type<any>, data?: any): void {
-    if (this.hsConfig == undefined || this.hsConfig.panelsEnabled[name]) {
-      this.HsPanelContainerService.create(panelComponent, data || {});
-    }
+  ngOnDestroy(): void {
+    this.end.next();
+    this.end.complete();
   }
-  ngOnInit(): void {
+
+  async ngOnInit(): Promise<void> {
     if (this.config) {
       this.hsConfig.update(this.config);
     }
     if (this.id) {
       this.hsConfig.setAppId(this.id);
     }
-    this.createPanel('tripPlanner', HsTripPlannerComponent, {});
-    this.createPanel('addData', HsAddDataComponent, {});
-    this.createPanel('draw', HsDrawComponent, {});
-    this.createPanel('search', HsSearchComponent, {});
-    this.createPanel('feature_table', HsFeatureTableComponent, {});
-    this.createPanel('saveMap', HsSaveMapComponent, {});
-    this.createPanel('language', HsLanguageComponent, {});
-    this.createPanel('info', HsQueryComponent, {});
-    this.createPanel('permalink', HsShareComponent, {});
-    this.createPanel('print', HsPrintComponent, {});
-    this.createPanel('measure', HsMeasureComponent, {});
-    this.createPanel('composition_browser', HsCompositionsComponent, {});
-    this.createPanel('legend', HsLegendComponent, {});
-    this.createPanel('layermanager', HsLayerManagerComponent, {});
-    this.createPanel('mapSwipe', HsMapSwipeComponent, {});
-
-    this.HsPanelContainerService.create(HsStylerComponent, {});
-
-    this.hsToolbarPanelContainerService.create(HsSearchToolbarComponent, {});
-    this.hsToolbarPanelContainerService.create(HsDrawToolbarComponent, {});
-    this.hsToolbarPanelContainerService.create(HsMeasureToolbarComponent, {});
-
-    this.HsOverlayPanelContainerService.create(HsGeolocationComponent, {});
-    this.HsOverlayPanelContainerService.create(HsInfoComponent, {});
-    this.HsOverlayPanelContainerService.create(
-      HsLayerManagerGalleryComponent,
-      {},
-    );
-    this.HsOverlayPanelContainerService.create(HsToolbarComponent, {});
-    this.HsOverlayPanelContainerService.create(HsQueryPopupComponent, {
-      service: this.hsQueryPopupService,
+
+    this.HsLayoutService.layoutElement = this.elementRef.nativeElement;
+
+    this.HsLayoutService.contentWrapper =
+      this.elementRef.nativeElement.querySelector('.hs-content-wrapper');
+
+    if (window.innerWidth < 600 && this.HsUtilsService.runningInBrowser()) {
+      const viewport = document.querySelector('meta[name="viewport"]');
+      viewport.setAttribute(
+        'content',
+        'width=device-width, initial-scale=0.6, maximum-scale=2, user-scalable=no',
+      );
+    }
```

Comment:
This was just copied but I mean yea  it might not be excatly correct in all the scenarios... but having a hslayers map as a fraction of a page which is not even 600px in width ist not something you really want Id say. Moreover this attribute sets meta tag which affects whole page not just hslayers element so I dont think this is an issue.